File: _release/bitmap/bitmap.test.cpp
#include "bitmap.cpp"

int main(){
	int million = 1000000;//百万
	int million10 = 10 * million;//千万
	unsigned long billion = 1000 * million;//10亿
	unsigned long large_number = 1 * billion;
    BitMap bitmap;
    for(unsigned long i=0;i<large_number;i++){
    	bitmap.set(i);
    }
    printf("exist:%u\n",bitmap.get(large_number-1));
    printf("not exist:%u\n",bitmap.get(large_number+2));
    //bitmap.set(123);

	return 1;
}

File: _release/bitmap/bitmap.test.php
<?php
require_once "bitmap.php";
ini_set ('memory_limit', '1024M');

$million = 1000000;
$million10 = 100 * $million;
$billion = 1000 * $million;
$large_number =  4 * $billion;

$bitmap = new BitMap();
$pos = 0;
for($i=0;$i<$large_number;$i++){
	if($i % $million == 0){
		echo ".";
	}
	if($i % $million10 == 0){
		echo "\n".date("Y-m-d H:i:s")."\n";
	}
	$bitmap->set($i);
	
}